From 09ed9fc3c1edbd961f352f724bc354c6d75b8088 Mon Sep 17 00:00:00 2001
From: AH-2004 <ahmedshuaib2004@gmail.com>
Date: Sat, 3 Jun 2023 01:51:22 +0500
Subject: [PATCH] Added variable in config file for refreshrate.

---
 config.def.h | 3 ++-
 dwm.c        | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/config.def.h b/config.def.h
index 9efa774..f15cd3c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -4,7 +4,8 @@
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
-static const int topbar             = 1;        /* 0 means bottom bar */
+static const int topbar = 1;                    /* 0 means bottom bar */
+static const int refreshrate = 60;              /* monitor refreshrate */
 static const char *fonts[]          = { "monospace:size=10" };
 static const char dmenufont[]       = "monospace:size=10";
 static const char col_gray1[]       = "#222222";
diff --git a/dwm.c b/dwm.c
index f1d86b2..c772234 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1172,7 +1172,7 @@ movemouse(const Arg *arg)
 			handler[ev.type](&ev);
 			break;
 		case MotionNotify:
-			if ((ev.xmotion.time - lasttime) <= (1000 / 60))
+			if ((ev.xmotion.time - lasttime) <= (1000 / refreshrate))
 				continue;
 			lasttime = ev.xmotion.time;
 
@@ -1326,7 +1326,7 @@ resizemouse(const Arg *arg)
 			handler[ev.type](&ev);
 			break;
 		case MotionNotify:
-			if ((ev.xmotion.time - lasttime) <= (1000 / 60))
+			if ((ev.xmotion.time - lasttime) <= (1000 / refreshrate))
 				continue;
 			lasttime = ev.xmotion.time;
 
-- 
2.40.1

